scio-examples/src/main/scala/com/spotify/scio/examples/cookbook/JoinExamples.scala

```diff
@@ -144,3 +144,74 @@ object HashJoinExamples {
     ()
   }
 }
+
+// ## Skewed join
+object SkewedJoinExamples {
+  def main(cmdlineArgs: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(cmdlineArgs)
+
+    import JoinUtil._
+
+    // Extract both sides as `SCollection[(String, String)]`s
+    val eventsInfo =
+      sc.bigQueryTable(Table.Spec(ExampleData.EVENT_TABLE)).flatMap(extractEventInfo)
+    val countryInfo =
+      sc.bigQueryTable(Table.Spec(ExampleData.COUNTRY_TABLE)).map(extractCountryInfo)
+
+    eventsInfo
+      // Skewed join is useful when LHS contains a subset of keys with high frequency, but RHS is
+      // too large to fit into memory. It uses Count Min Sketch to estimate those frequencies.
+      // Internally it identifies two groups of keys: "Hot" and the rest. Hot keys are joined using
+      // Hash Join and the rest with the regular join. There are 3 ways to identify
+      // the set of hot keys:
+      // 1) "threshold" as a cutoff frequency
+      // 2) "top percentage" to specify the maximum relative part of all keys can be considered hot
+      // 3) "top N" to specify the absolute number of hot keys
+      .skewedLeftOuterJoin(countryInfo, hotKeyThreshold = 100)
```

Comment:
That's nice.
Should we go a bit further and use the new API, with some explanation on how to pick good params for the CMS (epsilon, delta)?

Reply:
Agree!